### DISCUSSION TITLE
How can I use ff-only merging with jj?

### DISCUSSION THREAD

**Discussion by theoparis** [jj-vcs/jj#4743] [Q&A] [answered]
With git, there is a `--ff-only` merge option that will keep linear history without a merge commit.  I can't use rebasing because it prevents using signed commits (every commit gets marked as unverified after being merged).

Unfortunately,GitHub currently doesn't support this via the UI either. So I've resorted to using `git checkout main && git pull && git merge  --ff-only some/branch` to merge PRs in the meantime. However I haven't seen such a feature with jj... Is this something that can be supported or would I have to use `jj git clone --colocate` to be able to call `git merge`?

**Comment by yuja** [comment, marked as answer]
afaik, `merge --ff-only` isn't technically a merge, but moving a ref forward. The corresponding command in jj is `jj bookmark move --from .. --to ..` (to move ref) and `jj new` (to checkout the new revision.)

**Reply by theoparis** [reply]
Oh ok, thanks. I'm still getting used to how jj works. I ended up with something like this as an equivelent to `git merge --ff-only origin/add-github-workflow`:

```
jj bookmark move main@origin --to add-github-workflow@origin
jj git push --bookmark main
jj new main
```
